user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/security/ArtemisInternalAuthenticationProvider.java

```diff
@@ -0,0 +1,91 @@
+package de.tum.in.www1.artemis.usermanagement.security;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+import de.tum.in.www1.artemis.service.user.PasswordService;
+import de.tum.in.www1.artemis.usermanagement.service.user.UserCreationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.security.authentication.InternalAuthenticationServiceException;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+@ConditionalOnProperty(value = "artemis.user-management.use-external", havingValue = "false")
+public class ArtemisInternalAuthenticationProvider extends ArtemisAuthenticationProviderImpl implements ArtemisAuthenticationProvider {
+
+    private final Logger log = LoggerFactory.getLogger(ArtemisInternalAuthenticationProvider.class);
+
+    public ArtemisInternalAuthenticationProvider(UserRepository userRepository, PasswordService passwordService, UserCreationService userCreationService) {
+        super(userRepository, passwordService, userCreationService);
+    }
+
+    @Override
+    public User getOrCreateUser(Authentication authentication, String firstName, String lastName, String email, boolean skipPasswordCheck) {
+        final var password = authentication.getCredentials().toString();
+        final var optionalUser = userRepository.findOneByLogin(authentication.getName().toLowerCase());
+        final User user;
+        if (optionalUser.isEmpty()) {
+            user = userCreationService.createInternalUser(authentication.getName(), password, null, firstName, lastName, email, null, null, "en");
+        }
```

Comment:
[nit] If think we can remove the `optionalUser` variable by providing the default value in case the optional is empty.

```
final User user = 
  userRepository.findOneByLogin(authentication.getName().toLowerCase())
    .orElse(userCreationService.createInternalUser(authentication.getName(), password, null, firstName, lastName, email, null, null, "en"));

if (!skipPasswordCheck) {
  ..
}

return user;
```

Reply:
This would mean to also execute 
```
if (!skipPasswordCheck) {
  ..
}
```
when the user was just created which is not needed in my opinion. That's why I think about not adding this change. Let me know if you definitely think I should add it